src/sentry/models/group.py

```diff
@@ -201,10 +202,14 @@ def get_oldest_or_latest_event_for_environments(
         )
         dataset = Dataset.Transactions
     else:
+        if group.issue_category == GroupCategory.ERROR:
+            dataset = Dataset.Events
+        else:
+            dataset = Dataset.IssuePlatform
```

Comment:
for the `elif` I meant something like 
```
if ... == GroupCategory.PERFORMANCE:
    dataset = Dataset.Transactions
elif group.issue_category == GroupCategory.ERROR:
    dataset = Dataset.Events
else:
    dataset = Dataset.IssuePlatform
```

Reply:
The problem there is that `Transactions` has a different set of conditions applied to it than Events/IssuePlatform, which share the same ones.

Reply:
right i mean keeping the logic on line 198. it's just if the first two `if` fail it will always land on `Dataset.IssuePlatform` anyways. Just was suggesting to restructure the `if/else` statements

Reply:
Ok, reworked it so that performance issues are unable to make it to this step in general.